pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1075,3 +1101,98 @@ func (c *K8sOrchestrator) GetNodesForVolumes(ctx context.Context, volumeIDs []st
 	}
 	return volumeIDToNodeNames
 }
+
+// initNodeIDToNameMap performs all the operations required to initialize
+// the node ID to  name map. It also watches for node add, update & delete
+// operations, and updates the map accordingly.
+func initNodeIDToNameMap(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	// If use-node-id is set then do not initialize the informer
+	if !k8sOrchestratorInstance.IsFSSEnabled(ctx, common.UseCSINodeId) {
+		log.Debugf("Initializing node ID to node name map")
+		k8sOrchestratorInstance.nodeIDToNameMap = &nodeIDToNameMap{
+			RWMutex: &sync.RWMutex{},
+			items:   make(map[string]string),
+		}
+
+		// Set up kubernetes resource listener to listen events on Node
+		k8sOrchestratorInstance.informerManager.AddNodeListener(
+			func(obj interface{}) { // Add.
+				nodeAdd(obj)
+			},
+			nil,
+			func(obj interface{}) { // Delete.
+				nodeRemove(obj)
+			})
```

Comment:
Don't we need nodeUpdate listener, as mentioned in the comments above the function definiton

Reply:
I will update the comment, but we do not need nodeUpdate listener, since we are constructing the nodeID to name map here, which are available when node is created/ added. So, I am not sure we need the update function.